### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes kiltprotocol/ticket#627 ; fixes kiltprotocol/ticket#628 ; fixes kiltprotocol/ticket#629
The app has been sending messages that could not be loaded by both itself and the demo client.
It also got stuck in a loop after screen lock as it constantly tried to unlock the identity, but would reset when doing so.

## How to test:
Run our new checklist against this build.

## Checklist:

- [x] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: src/components/MnemonicCreation.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import { View, Text, TextInput } from 'react-native'
```
Comment (rflechtner) [inline src/components/MnemonicCreation.tsx:2]: unused import

File: src/components/MnemonicCreation.tsx [hunk 2]
```diff
@@ -106,7 +106,7 @@ class MnemonicCreation extends React.Component<Props, State> {
                 returnKeyType="done"
                 multiline={true}
                 numberOfLines={3}
-                maxLength={80}
+                maxLength={100}
```
Comment (Dudleyneedham) [inline src/components/MnemonicCreation.tsx:109]: Why was this changed?

File: src/containers/AppRoot.tsx [hunk 3]
```diff
@@ -121,11 +124,20 @@ class AppRoot extends React.Component<Props> {
   }
 
   async promptAndSetDecryptedIdentity(): Promise<void> {
-    console.info('[ENCRYPTION] Decrypting and setting identity in store')
-    const { setIdentityInStore } = this.props
-    const identityDecrypted = await promptUserAndGetIdentityDecrypted()
-    setIdentityInStore(identityDecrypted)
-    console.info('[ENCRYPTION] OK identity set in store')
```
Comment (Dudleyneedham) [inline src/containers/AppRoot.tsx:128]: Is this because of the `async` nature of the identity now?

File: src/services/service.messaging.ts [hunk 4]
```diff
@@ -105,7 +105,7 @@ function isMessageNew(
   oldMessages: IMessageMapById,
   messageId: Message['messageId']
 ): boolean {
-  return !oldMessages[messageId]
+  return !!messageId && !oldMessages[messageId]
```
Comment (Dudleyneedham) [inline src/services/service.messaging.ts:108]: Why is this done?

File: src/components/MnemonicCreation.tsx (hunk 2 above)
Comment (rflechtner) [reply src/components/MnemonicCreation.tsx:109]: it was to short for most seed phrases :D

Comment (Dudleyneedham) [reply src/components/MnemonicCreation.tsx:109]: Cool :)

File: src/containers/AppRoot.tsx (hunk 3 above)
Comment (rflechtner) [reply src/containers/AppRoot.tsx:128]: no, it just didn't work, at least on my device. that fixes the finger scan popup constantly reappearing (see ticket)

File: src/services/service.messaging.ts (hunk 4 above)
Comment (rflechtner) [reply src/services/service.messaging.ts:108]: that's just a lint fix - messageId was potentially undefined

File: src/containers/AppRoot.tsx (hunk 3 above)
Comment (rflechtner) [reply src/containers/AppRoot.tsx:128]: I didn't make it async or anything, I'm just avoiding that the trigger fires twice in quick succession and blocks everything
